async-profiler.yaml

```diff
@@ -0,0 +1,44 @@
+package:
+  name: async-profiler
+  version: 3.0
+  epoch: 0
+  description: A low overhead sampling profiler for Java
+  copyright:
+    - license: Apache-2.0
+
+environment:
+  contents:
+    packages:
+      - bash
+      - build-base
+      - openjdk-11
+      - openjdk-11-default-jvm
+      - wolfi-base
+  environment:
+    JAVA_HOME: /usr/lib/jvm/java-11-openjdk
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/async-profiler/async-profiler
+      tag: v${{package.version}}
+      expected-commit: 4e441b4024a5873a5764a1e8b9f0bb25ad997fbf
+
+  - runs: |
+      make all
```

Comment:
It looks like the upstream `Makefile` does _not_ use `CFLAGS` and `CXXFLAGS` from the environment which is not ideal, unsure if we should patch that out: https://github.com/async-profiler/async-profiler/blob/4e441b4024a5873a5764a1e8b9f0bb25ad997fbf/Makefile#L11-L12